xmtp_pppp/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+mod encryption;
+pub mod topic;
+
+use prost::Message as ProstMessage;
+use xmtp_proto::xmtp::message_contents::{
+    private_preferences_payload::Version as PrivatePreferencesVersion, Ciphertext,
+    PrivatePreferencesPayload,
+};
+
+use crate::encryption::{decrypt_ciphertext, encrypt_to_ciphertext};
+
+pub fn encrypt_message(
+    private_key: &[u8], // secp256k1 private key
+    message: &[u8],     // any byte array
+    topic: String,      // the topic you are encrypting for
```

Comment:
I am taking the topic in as an argument and using it as associated data in the encryption. I could omit the associated data entirely, since we don't have anything great to use here, but it does explicitly make it impossible to copy messages from one topic to another. Given that the topic is tied to the private key, that's probably not a real concern.

Reply:
It does mean we can't change/migrate the topic format in the future, as it's coupled directly to the encryption. Would it make sense to use the public key here, which would theoretically make rotations easier to support?